timesketch/ui/graphs/cytoscape-settings.component.spec.ts

```diff
@@ -0,0 +1,137 @@
+import {Component, ViewChild} from '@angular/core'
+import {TestBed, ComponentFixture} from '@angular/core/testing'
+import {BrowserDynamicTestingModule, platformBrowserDynamicTesting} from '@angular/platform-browser-dynamic/testing'
+import {BrowserModule, By} from '@angular/platform-browser'
+import {FormsModule} from '@angular/forms'
+
+import {CytoscapeComponent} from './cytoscape.component'
+import {testCytoscapeOptions, getOptionFromCytoscape} from './cytoscape.component.spec'
+import {CytoscapeSettingsComponent} from './cytoscape-settings.component'
+
+describe('CytoscapeComponent', () => {
+  beforeAll(() => {
+    TestBed.resetTestEnvironment()
+    TestBed.initTestEnvironment(
+      BrowserDynamicTestingModule, platformBrowserDynamicTesting(),
+    )
+  })
+
+  const remainingOptions = {...testCytoscapeOptions}
+
+  let fixture: ComponentFixture<SettingsTestComponent>
+  beforeEach(async () => {
+    TestBed.configureTestingModule({
+      imports: [BrowserModule, FormsModule],
+      declarations: [CytoscapeComponent, CytoscapeSettingsComponent, SettingsTestComponent],
+    })
+    fixture = TestBed.createComponent(SettingsTestComponent)
+    // prevent 'Error: Timeout - Async callback was not invoked within timeout':
+    fixture.componentInstance.cytoscapeComponent['container'] = undefined
+    fixture.detectChanges()
+    await fixture.whenStable()
+  })
+
+  describe('checkboxes', () => {
+    it('should work correctly', () => {
```

Comment:
I would create a local list of all options here that are represented by a checkbox. Then you can iterate this list and see that all options can be toggled on the UI.

To make sure that people do not forget to update the tests when new options are introduced, I would create a second test case that checks that all checkboxes are in your local list. I.e. two tests:

before(('...') => {
  const booleanOptions = ['a', 'b', 'c'];
  it('allows toggling options', () => {
    // Checks that all elements in booleanOptions are present in the template.
  });
  if('does not specify additional boolean options', () => {
    // Checks that all checkboxes in the template are present in booleanOptions.
  });
});

WDYT?

Reply:
There are already assertions to ensure that all options are tested. Applying this suggestion would introduce some duplication, maybe even worth adding for readability. But I would rather not spend more time on it, there are other priorities.